contrib/pkg/utils/aws/aws.go

```diff
@@ -1,16 +1,216 @@
 package aws
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
 	"github.com/openshift/hive/contrib/pkg/utils"
+	"github.com/openshift/hive/pkg/awsclient"
 	"github.com/openshift/hive/pkg/constants"
+
 	log "github.com/sirupsen/logrus"
 	ini "gopkg.in/ini.v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	// PrivateLinkHubAcctCredsName is the name of the AWS PrivateLink Hub account credentials Secret
+	// created by the "hiveutil awsprivatelink enable" command
+	PrivateLinkHubAcctCredsName = "awsprivatelink-hub-acct-creds"
+
+	// PrivateLinkHubAcctCredsLabel is added to the AWS PrivateLink Hub account credentials Secret
+	// created by the "hiveutil awsprivatelink enable" command and
+	// referenced by HiveConfig.spec.awsPrivateLink.credentialsSecretRef.
+	PrivateLinkHubAcctCredsLabel = "hive.openshift.io/awsprivatelink-hub-acct-credentials"
+)
+
+func GetAWSClientsByRegion(regions map[string]bool) (map[string]awsclient.Client, error) {
+	awsClientsByRegion := make(map[string]awsclient.Client)
+	for region := range regions {
+		awsClients, err := awsclient.NewClientFromSecret(nil, region)
+		if err != nil {
+			return awsClientsByRegion, err
+		}
+		awsClientsByRegion[region] = awsClients
+	}
+
+	return awsClientsByRegion, nil
+}
+
+func FindVpcInInventory(vpcId string, inventory []hivev1.AWSPrivateLinkInventory) (int, bool) {
+	var endpointVpcIdx int
+	var found bool
+
+	for i, endpointVpc := range inventory {
+		if vpcId == endpointVpc.AWSPrivateLinkVPC.VPCID {
+			endpointVpcIdx = i
+			found = true
+			break
+		}
+	}
+
+	return endpointVpcIdx, found
```

Comment:
or
```suggestion
	for i, endpointVpc := range inventory {
		if vpcId == endpointVpc.AWSPrivateLinkVPC.VPCID {
			return i, true
		}
	}
	return -1, false
```

Reply:
Go's implicit return style was used here. I switched to the explicit style later but forgot to clean this function up ... 

Fixed.